cloudformation/cloudformation.yaml

```diff
@@ -8,7 +8,109 @@ Parameters:
             - PROD
             - CODE
         Default: CODE
+Mappings:
+  EnvironmentBucketMap:
+    CODE:
+      "logs": "fulfilment-s3-logs-code"
+      "salesforce": "fulfilment-salesforce-backup-code"
+      "zuora": "fulfilment-zuora-staging-code"
+      "export": "fulfilment-export-code"
+    PROD:
+      "logs": "fulfilment-s3-logs-prod"
+      "salesforce": "fulfilment-salesforce-backup-prod"
+      "zuora": "fulfilment-zuora-staging-prod"
+      "export": "fulfilment-export-prod"
 Resources:
+  FulfilmentAccessLogBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName:
+        Fn::FindInMap:
+          - EnvironmentBucketMap
+          - Ref: Stage
+          - "logs"
+      AccessControl: LogDeliveryWrite
+      LoggingConfiguration:
+        DestinationBucketName:
+          Fn::FindInMap:
+            - EnvironmentBucketMap
+            - Ref: Stage
+            - "logs"
+        LogFilePrefix: "logs-logs/"
+      VersioningConfiguration:
+        Status: "Enabled"
+  SalesForceBackupBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName:
+        Fn::FindInMap:
+          - EnvironmentBucketMap
+          - Ref: Stage
+          - "salesforce"
+      AccessControl: Private
+      LifecycleConfiguration:
+        Rules:
+          - Id: DeleteOldFilesSF
+            Prefix: ""
+            Status: Enabled
+            ExpirationInDays: 21
+      LoggingConfiguration:
+        DestinationBucketName:
+          Fn::FindInMap:
+            - EnvironmentBucketMap
+            - Ref: Stage
+            - "logs"
+        LogFilePrefix: "fulfilment-salesforce-backup-_[STAGE]/"
+      VersioningConfiguration:
+        Status: "Enabled"
+  ZuoraExportBucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName:
+        Fn::FindInMap:
+          - EnvironmentBucketMap
+          - Ref: Stage
+          - "zuora"
+      AccessControl: Private
+      LifecycleConfiguration:
+        Rules:
+          - Id: DeleteOldFiles
+            Prefix: ""
+            Status: Enabled
+            ExpirationInDays: 21
+      LoggingConfiguration:
+        DestinationBucketName:
+          Fn::FindInMap:
+            - EnvironmentBucketMap
+            - Ref: Stage
+            - "logs"
+        LogFilePrefix: "fulfilment-salesforce-backup-_[STAGE]/"
```

Comment:
This should be "fulfilment-export-_[STAGE]/"

Reply:
does !Sub fulfilment-salesforce-backup-${Stage}/  not work for this?

Reply:
No, this is totally wrong, I need to do a map lookup as s3 buckets can't have caps in their names.